src/thx/fp/List.hx

```diff
@@ -16,7 +16,15 @@ abstract List<A>(ListImpl<A>) from ListImpl<A> to ListImpl<A> {
 
   @:from
   inline static public function fromArray<A>(arr : Array<A>) : List<A>
-    return arr.reduceRight.fn(bin(_1, _0), empty());
+    return arr.reduceRight(
+      function(memo,next){
+        return switch(memo){
```

Comment:
Or if you want to get really fancy, `arr.reduceRight(thx.fp.Functions.flip(Cons), empty())`